libraries/SPI/SPI.cpp

```diff
@@ -74,13 +75,17 @@ void SPIClass::setHwCs(bool use) {
 }
 
 void SPIClass::beginTransaction(SPISettings settings) {
-    while(SPI1CMD & SPIBUSY) {}
+    while(inTransaction || SPI1CMD & SPIBUSY) {
```

Comment:
I want to read sensors data from an interrupt or ticker, but as the interrupt may come while writing to the LCD or even while reading another sensor, I was looking for a way to protect the transaction.
The idea is that the last called interrupt handler will yield() to allow the previous one to complete it's SPI communication.
Why yield is not interrupt safe? Were can I read more about it?